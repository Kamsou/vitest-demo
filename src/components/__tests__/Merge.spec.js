import { test, expect } from "vitest";
import { deepMerge } from "@/utils/index.js";

test("shallow merge", () => {
  const merged = deepMerge({ name: "Camille" }, { github: "Kamsou" });

  expect(merged).toMatchSnapshot();
});

test("shallow merge with arrays", () => {
  const merged = deepMerge(["vue", "react"], ["svelte", "angular"]);

  expect(merged).toEqual(["vue", "react", "svelte", "angular"]);
});

test("throws errors on merging two different types", () => {
  expect(() => deepMerge(["foo", "bar"], { foo: "bar" })).toThrowError(
    "Cannot merge two different types"
  );
});
