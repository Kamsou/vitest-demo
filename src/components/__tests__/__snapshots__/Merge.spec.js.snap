// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`shallow merge 1`] = `
{
  "github": "Kamsou",
  "name": "Camille",
}
`;
